staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -48,3 +48,58 @@ type AdmissionPluginConfiguration struct {
 	// +optional
 	Configuration *runtime.Unknown `json:"configuration"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConnectivityServiceConfiguration provides versioned configuration for network proxy clients.
+type ConnectivityServiceConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// connectionServices contains a list of network proxy client configurations
+	ConnectionServices []ConnectionService `json:"connectionServices"`
+}
+
+// ConnectionService provides the configuration for a single network proxy client.
+type ConnectionService struct {
+	// name is the name of the connectivity service.
+	// Currently supported values are "Master" and "Cluster"
```

Comment:
code below suggests Etcd works.

Reply:
I will fix this. To be clear, the KEP specifies this should work for etcd. However I am trying to break it into pieces. So the change which actually enables that is still coming.